### ISSUE TITLE
Compile issue for version 1.2.0 on Magento 2.1.9 ?

### ISSUE THREAD

**Issue by EmilyPepperman** [amzn/amazon-payments-magento-2-plugin#154]
Expected to update to version of Amazon Pay 1.2.0 smoothly

When we try to setup:di:compile we get this error:
[RuntimeException]
Source class "\amazonClient" for "amazonClientLogger" generation does not exist.

"amzn/amazon-payments-magento-2-plugin":"1.2.0",

* Magento version: 2.1.9
* Magento Edition: Community
* JavaScript Console: Please add a screenshot of the Chrome JavaScript console to speed up the process
* Shop URL : dev.divegearexpress.com

**Comment by danielneu** [comment]
Hi @EmilyPepperman 

this is strange, the compilation is part of our release process and I just double checked it - no such issue for me.

I also set up a fresh 2.1.19 shop with version 1.1.5 of our module installed and then upgraded to 1.2.0 - no issue.

The amazonClientLogger is simply Monolog: https://github.com/amzn/amazon-payments-magento-2-plugin/blob/a756d4d99c17b5aaddcea3a9648fecdd0d3e5e1d/src/Core/etc/di.xml#L52

amazonClient is our Client.php delivered by our SDK.
Can you see our SDK in vendor/amzn/?

Best wishes 
Daniel

**Comment by EmilyPepperman** [comment]
So we are not positive where the issue is... but we can't compile if have the Mirasvit latest version of their Followup Email extension https://mirasvit.com/magento-2-extensions/follow-up-email.html and Amazon Pay extension enabled. So has to be some conflict between the two extensions... for now... we have rolled back the version of Mirasvit and do have 1.2.0 of Amazon installed.

**Comment by EmilyPepperman** [comment]
I really need to be able to update Miravit extension because the latest release contains 2.2 compatibility and we are trying to get updated. 
I am getting this error while trying to bin/magento setup:di:compile

 [RuntimeException]
  Source class "\amazonClient" for "amazonClientLogger" generation does not exist.

Yes the SDK is in vendor/amzn

**Comment by EmilyPepperman** [comment]
in setup_module I have
Amazon_Core | 1.2.0 | 1.2.0
Amazon_Login | 1.2.0 | 1.2.0
Amazon_Payment | 1.11.0 | 1.11.0

**Comment by EmilyPepperman** [comment]
So we have reverting back to an older version of Mirasvit - Follow up email extension and the issue is gone... I wonder if anyone also using Mirasvit's extension is having the same issue?

**Comment by EmilyPepperman** [comment]
ok @danielneu Mirasvit has confessed they see the issue now... and it is their fault. 

Hello Emily

We found a problem, since version 1.1.0 of the Mirasvit Email extension we added new module Mirasvit Event, which for some reason causes this error when Amazon Payment extension installed.

Currently, we are trying to investigate this issue and solve.

We will contact you upon completion,
Thank you,

best regards, 
Alexey
